packages/frontend/src/pages/ParkingSpace/index.tsx

```diff
@@ -35,21 +36,62 @@ const ParkingSpace = () => {
 }
 
 const ParkingSpaceTabs = (props: { listingId: number }) => {
-  const [selectedTab, setSelectedTab] = useState('1')
   const { data } = useParkingSpaceListing({
     id: props.listingId,
   })
 
+  let initialTab = '1'
+  if (data.status === ListingStatus.Assigned && data.offers.length > 0) {
+    initialTab = String(data.offers[0].id)
+  }
+
+  const [selectedTab, setSelectedTab] = useState(initialTab)
```

Comment:
You can move this code into the useState initializer:

`useState(() => {...})`
`useState(myPreDefinedFn)`